### PR TITLE
SB-36: video view

### PR DESCRIPTION
**Pull request by eccentricnode**
## Feature/Problem

Sermon View needed refactoring done on the template, logic setup, so the data is served properly, and some styling and layout refinements.

## Solution

Refactored template, setup logic so data is served properly, and changed styling and layout as needed.

## Areas of Impact

Sermon View 

## UI Images

Annotation 2019-12-19 201950

## Manual Testing

- Verify CI passes

### CODE REVIEW

File: client/apps/user-portal/src/app/sermon-view/sermon-view.component.html [hunk 1]
```diff
@@ -1,20 +1,16 @@
-<div *ngIf="sermon$ | async as sermon">
+<div *ngIf="sermon$ | async as sermon"> 
+  <!-- {{sermon | json}} -->
```
Comment (VicAv99) [inline client/apps/user-portal/src/app/sermon-view/sermon-view.component.html]:
```suggestion
```

File: client/apps/user-portal/src/app/sermon-view/sermon-view.component.html [hunk 2]
```diff
@@ -1,20 +1,16 @@
...
-        height="625"
-        frameborder="0"
-        [title]="sermon.title"
-        webkitallowfullscreen
-        mozallowfullscreen
-        allowfullscreen>
-      </iframe>
+      <ng-container *ngIf="media$ | async as media">
+        <div class="embed-responsive embed-responsive-16by9" *ngFor="let m of media">
+          <div *ngIf="m.embedCode" class="embed-responsive-item" [innerHtml]="m.embedCode"></div>
```
Comment (VicAv99) [inline client/apps/user-portal/src/app/sermon-view/sermon-view.component.html]: Isn't it `[innerHTML]`?

File: client/apps/user-portal/src/app/sermon-view/sermon-view.component.ts [hunk 3]
```diff
@@ -1,62 +1,119 @@
...
-    { title: 'SAVE', icon: 'cloud_download' },
-    { title: 'LISTEN', icon: 'graphic_eq' },
-    { title: 'NOTES', icon: 'notes' }
-  ];
+  sermon$: Observable<Sermon> = this.sermonsFacade.selectedSermon$;
+  speaker$: Observable<Speaker[]> = this.speakersFacade.allSpeakers$;
+  media$: Observable<Partial<Media[]>> = this.mediaFacade.allMedia$;
+  tags$: Observable<Tag[]> = this.tagsFacade.allTags$;
+
+  test = 'AUDIO';
```
Comment (VicAv99) [inline client/apps/user-portal/src/app/sermon-view/sermon-view.component.ts]: Is this supposed to stay here?

File: client/libs/sermons/src/lib/sermons/sermons.component.html [hunk 4]
```diff
@@ -59,8 +59,8 @@
       </ng-container>
 
       <tr mat-header-row *matHeaderRowDef="displayedColumns; sticky: true"></tr>
-      <tr mat-header-row *matHeaderRowDef="spacerColumns; sticky: true"></tr>
-      <tr mat-row *matRowDef="let row; columns: displayedColumns;" class="mat-list-option"></tr>
+      <tr mat-header-row *matHeaderRowDef="spacerColumns; sticky: true" class="example-second-header-row"></tr>
```
Comment (VicAv99) [inline client/libs/sermons/src/lib/sermons/sermons.component.html]:
```suggestion
      <tr mat-header-row *matHeaderRowDef="spacerColumns; sticky: true"></tr>
```

File: server/config.yaml [hunk 5]
```diff
@@ -1 +1 @@
-endpoint: http://0.0.0.0:8080
+endpoint: http://127.0.0.1:8080
```
Comment (VicAv99) [inline server/config.yaml]:
```suggestion
endpoint: http://0.0.0.0:8080
```

**Comment by VicAv99** [comment]
The build is failing

File: client/apps/user-portal/src/app/sermon-view/sermon-view.component.ts [hunk 6]
```diff
@@ -1,62 +1,119 @@
+import { ActivatedRoute, Router } from '@angular/router';
 import { Component, OnInit } from '@angular/core';
-import { Router } from '@angular/router';
-import { Observable, of } from 'rxjs';
-import { Sermon } from '@sb/core-data';
 import { DomSanitizer } from '@angular/platform-browser';
 
+import { Observable } from 'rxjs';
+
+import { Media, MediaTypes, Sermon, Speaker, Tag } from '@sb/core-data';
```
Comment (VicAv99) [inline client/apps/user-portal/src/app/sermon-view/sermon-view.component.ts]: Have you pulled in latest? The `MediaTypes` import should be `MediaType`.

File: client/apps/user-portal/src/app/sermon-view/sermon-view.component.ts [hunk 7]
```diff
@@ -1,62 +1,119 @@
...
+      const removePTags = m.embedCode.split('<p')[0];
+
+      return {...m, embedCode: this.sanitizeHtml(removePTags)};
+    });
   }
 
-  getVideoUrl(media: any[]) {
-    const videoMedia = media.find((m) => m.type === 'VIDEO');
+  grabSermon(currentSermonId: string) {
+    return this.sermonsFacade.selectSermon(currentSermonId);
```
Comment (VicAv99) [inline client/apps/user-portal/src/app/sermon-view/sermon-view.component.ts]:
```suggestion
    this.sermonsFacade.selectSermon(currentSermonId);
```

File: client/apps/user-portal/src/app/sermon-view/sermon-view.component.ts [hunk 8]
```diff
@@ -1,62 +1,119 @@
...
+        return this.listenToAudio();
+      }
+      case MediaTypes.PDF: {
+        return this.downloadNotes();
+      }
+      case MediaTypes.VIDEO: {
+        return this.downloadVideo();
+      }
+      default: {
+        // Do nothing
```
Comment (VicAv99) [inline client/apps/user-portal/src/app/sermon-view/sermon-view.component.ts]:
```suggestion
        break;
```

File: client/apps/user-portal/src/app/sermon-view/sermon-view.component.html (hunk 2 above)
Comment (eccentricnode) [reply client/apps/user-portal/src/app/sermon-view/sermon-view.component.html]: That's weird. It work anyway. lol

**Pull request merged**